src/openassetio-core/tests/hostApi/ManagerTest.cpp

```diff
@@ -313,6 +313,266 @@ SCENARIO("Resolving entities") {
+
+      openassetio::BatchElementError expected{
+          openassetio::BatchElementError::ErrorCode::kMalformedEntityReference, "Error Message"};
+
+      // With success callback side effect
+      REQUIRE_CALL(*mockManagerInterfacePtr,
+                   register_(refs, singleTraitsDatas, context, hostSessionPtr, _, _))
+          .LR_SIDE_EFFECT(_6(0, expected));
+
+      WHEN("singular register is called with default errorPolicyTag") {
+        THEN("an exception is thrown") {
+          CHECK_THROWS_MATCHES(manager->register_(ref, singleTraitsData, context),
+                               openassetio::MalformedEntityReferenceBatchElementException,
+                               Catch::Message("Error Message"));
+        }
+      }
+      WHEN("singular register is called with kException errorPolicyTag") {
+        THEN("an exception is thrown") {
+          CHECK_THROWS_MATCHES(
+              manager->register_(ref, singleTraitsData, context,
+                                 hostApi::Manager::BatchElementErrorPolicyTag::kException),
+              openassetio::MalformedEntityReferenceBatchElementException,
+              Catch::Message("Error Message"));
+        }
+      }
+      WHEN("singular register is called with kVariant errorPolicyTag") {
+        std::variant<openassetio::BatchElementError, openassetio::EntityReference> actual =
+            manager->register_(ref, singleTraitsData, context,
+                               hostApi::Manager::BatchElementErrorPolicyTag::kVariant);
+        THEN("returned variant contains the expected BatchElementError") {
+          CHECK(std::holds_alternative<openassetio::BatchElementError>(actual));
+          const auto& actualVal = std::get<openassetio::BatchElementError>(actual);
+          CHECK(expected == actualVal);
+          // TODO(EM): BatchElementError is currently copied, but we
+          //  may switch this to move semantics.
+          // https://github.com/OpenAssetIO/OpenAssetIO/issues/858
+          CHECK_FALSE(expected.message.data() == actualVal.message.data());
+        }
+      }
+    }
+    GIVEN(
+        "manager plugin will encounter entity-specific errors when next registering multiple "
+        "EntityReferences") {
+      const openassetio::EntityReferences refs = {openassetio::EntityReference{"ref1"},
+                                                  openassetio::EntityReference{"ref2"},
+                                                  openassetio::EntityReference{"ref3"}};
+
+      const openassetio::EntityReference expectedValue2{"expectedRef2"};
+      const openassetio::BatchElementError expectedError0{
+          openassetio::BatchElementError::ErrorCode::kMalformedEntityReference,
+          "Malformed Mock Error🤖"};
+      const openassetio::BatchElementError expectedError1{
+          openassetio::BatchElementError::ErrorCode::kEntityAccessError,
+          "Entity Access Error Message"};
+
+      // With success callback side effect
```

Comment:
[fix]
```suggestion
      // With mixed callback side effect
```
perhaps?

Reply:
Aye, fixed as part of : https://github.com/OpenAssetIO/OpenAssetIO/pull/879/commits/b17da05a778f733c17f627edce68b13b4a4a8057